pkg/resources/grant_helpers.go

```diff
@@ -54,12 +55,20 @@ type grant struct {
 // splitGrantID takes the <db_name>|<schema_name>|<view_name>|<privilege> ID and
 // returns the object name and privilege.
 func splitGrantID(v string) (string, string, string, string, error) {
-	arr := strings.Split(v, "|")
-	if len(arr) != 4 {
+
+	reader := csv.NewReader(strings.NewReader(v))
+	reader.Comma = '|'
+
+	lines, err := reader.ReadAll()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if len(lines) != 4 {
```

Comment:
lines is a `[][]string` I think you should check that `len(lines) == 1 && len(lines[0]) == 4`